lighthouse-core/audits/dobetterweb/password-inputs-can-be-pasted-into.js

```diff
@@ -23,15 +23,19 @@ class PasswordInputsCanBePastedIntoAudit extends Audit {
   }
 
   /**
-   * @param {!Artifacts} artifacts
-   * @return {!AuditResult}
+   * @param {LH.Artifacts} artifacts
+   * @return {LH.Audit.Product}
    */
   static audit(artifacts) {
     const passwordInputsWithPreventedPaste = artifacts.PasswordInputsWithPreventedPaste;
 
-    const items = passwordInputsWithPreventedPaste.map(input => ({
-      node: {type: 'node', snippet: input.snippet},
-    }));
+    /** @type {Array<{node: LH.Audit.DetailsRendererNodeDetailsJSON}>} */
+    const items = [];
+    passwordInputsWithPreventedPaste.forEach(input => {
```

Comment:
😢 TS really can't deal with a map + assertion?

Reply:
it loses the type `'node'` (it widens to type `string`) :( The heuristics are wonky on widening like that, but people do want very different things at times (even we would probably usually want `string` in this case if it weren't for wanting to tell the difference between details types).

It's basically this or put a cast in there in addition to the declared type above, I believe, so this seemed better.